packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -67,6 +84,29 @@ Each column that is sorted will have an appropriate icon displayed in the header
 
 Each column that is sorted ascending will have `aria-sort="ascending"` set on the header, and each column that is sorted descending will have `aria-sort="descending"` set on the header.
 
+### Interactive Sorting UX
+
+The UX/ interactions for interactive sorting are captured in [#885: UX Design for basic column/row interactions](https://github.com/ni/nimble/issues/885).  
+To summarize:
+
+Single-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were also sorted become unsorted (i.e. `currentSortDirection` => none/ `undefined`, `currentSortIndex` => `undefined` for those columns)
+-   The clicked column gets a `currentSortIndex` of `0` when sorted, and `undefined` if it's transitioning back to unsorted.
+
+Shift-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were already sorted remain sorted (their `currentSortIndex` values will be normalized to `0..(n-1)`, for `n` sorted columns)
+-   If the clicked column is transitioning to unsorted, its `currentSortIndex` will be set to `undefined`
+-   If the clicked column is transitioning to being sorted, it will get a `currentSortIndex` of `0` if no other columns were already sorted, or `n` if there were `n` other columns already sorted
+
+If sorting is enabled for a column, sorting menu items also appear in the column header menu:
+![Sorting via Column Header Menu](./spec-images/HeaderMenuSorting.png)  
+If sorting is disabled for a column, and no other enabled items are in the column header menu, the column header menu button will not appear.  
+Updating sorting via the menu will always unsort any other columns that were already sorted, even when the current column is being unsorted too.
+
+For columns with `sorting-disabled` set to true, clicking/Shift-clicking the column header will not affect the sort state, the sort menu options will not appear in the column header menu, and programmatic sorting is disabled (setting `sortIndex`/`sortDirection` will have no effect if `sorting-disabled` is also `true`).
```

Comment:
Will currentSortIndex and currentSortDirection have particular state in this mode? I expect undefined for both?